packages/adblocker/test/lists.test.ts

```diff
@@ -9,9 +9,91 @@
 import { expect } from 'chai';
 import 'mocha';
 
-import { f, generateDiff, getLinesWithFilters, mergeDiffs, parseFilters } from '../src/lists.js';
+import {
+  f,
+  generateDiff,
+  getLinesWithFilters,
+  mergeDiffs,
+  parseFilters,
+  detectFilterType,
+  FilterType,
+} from '../src/lists.js';
 import Config from '../src/config.js';
 
+const ADGUARD_FILTERS = [
+  // https://adguard.com/kb/general/ad-filtering/create-own-filters/#html-filtering-rules
+  'example.org$$script[data-src="banner"]',
+  '$@$script[tag-content="banner"]',
+  // CSS https://adguard.com/kb/general/ad-filtering/create-own-filters/#cosmetic-css-rules
+  'example.com#$#body { background-color: #333!important; }',
+  'example.com#@$#.textad { visibility: hidden; }',
+  // JavaScript https://adguard.com/kb/general/ad-filtering/create-own-filters/#javascript-rules
+  '#%#window.__gaq = undefined;',
+  '#@%#window.__gaq = undefined;',
+  // Extended CSS https://adguard.com/kb/general/ad-filtering/create-own-filters/#extended-css-selectors
+  'example.org#?#div:has(> a[target="_blank"][rel="nofollow"])',
+  'example.com#$?#h3:contains(cookies) { display: none!important; }',
+  'example.net#@?#.banner:matches-css(width: 360px)',
+];
+
+describe('#detectFilterType', () => {
+  const networkFilter = 'foo.com';
+  const cosmeticFilter = '###test';
+
+  it('detects NETWORK filters', () => {
+    expect(detectFilterType(networkFilter)).to.equal(FilterType.NETWORK);
+  });
+
+  it('detects COSMETIC filters', () => {
+    expect(detectFilterType(cosmeticFilter)).to.equal(FilterType.COSMETIC);
+  });
+
+  it('detects NON_SUPPORTED filters', () => {
+    expect(detectFilterType('')).to.equal(FilterType.NOT_SUPPORTED);
+    expect(detectFilterType(`a`)).to.equal(FilterType.NOT_SUPPORTED);
+    expect(detectFilterType('! some comment')).to.equal(FilterType.NOT_SUPPORTED);
+    expect(detectFilterType(`!${networkFilter}`)).to.equal(FilterType.NOT_SUPPORTED);
+    expect(detectFilterType(`!${cosmeticFilter}`)).to.equal(FilterType.NOT_SUPPORTED);
+    expect(detectFilterType(`!${cosmeticFilter}`)).to.equal(FilterType.NOT_SUPPORTED);
+    expect(detectFilterType(`[Adblock]`)).to.equal(FilterType.NOT_SUPPORTED);
+    for (const adguardFilter of ADGUARD_FILTERS) {
+      expect(detectFilterType(adguardFilter)).to.equal(FilterType.NOT_SUPPORTED, adguardFilter);
+    }
+  });
+
+  context('with extendedNonSupportedTypes option', () => {
+    const subject = (filter: string) =>
```

Comment:
"subject" is a quite generic name. Since all the tests have the form

expect(subject(filter)).to.equal(type);

could be more readable to express them as

expectType(filter, type)

where 
const expectType = (filter: string, type) =>
  expect(detectFilterType(filter, { extendedNonSupportedTypes: true }), type);

Reply:
yes, `subject` is a generic name, often used by convention in BDD testing :) it's only a helper function so we don't repeat the long function call

Reply:
but I like the suggestion, applied!